config/locales/server.en.yml

```
@@ -1763,6 +1762,7 @@ en:
     email_subject: "Customizable subject format for standard emails. See <a href='https://meta.discourse.org/t/customize-subject-format-for-standard-emails/20801' target='_blank'>https://meta.discourse.org/t/customize-subject-format-for-standard-emails/20801</a>"
     detailed_404: "Provides more details to users about why they can’t access a particular topic. Note: This is less secure because users will know if a URL links to a valid topic."
     enforce_second_factor: "Require users to enable two-factor authentication before they can access the Discourse UI. Select 'all' to enforce it to all users. Select 'staff' to enforce it to staff users only. This setting does not affect API or 'DiscourseConnect provider' authentication."
+    enforce_second_factor_on_external_auth: "Require users to use two-factor authentication even when they authenticate using external providers."
```

```suggestion
    enforce_second_factor_on_external_auth: "Require users to use two-factor authentication at all times. When enabled this will prevent users logging in with external authentication methods like social logins if they have two-factor authentication enabled. When disabled users will only need to confirm their two-factor authentication when logging in with a username and password. Also see the `enforce_second_factor` setting.`
```

Some suggestions because this is kind of a weird setting that we are adding, it needs to be super clear what it does.

---

Actually I think we should change the `enforce_second_factor` description too:

> Require users to enable two-factor authentication before they can access the Discourse UI. This setting does not affect API or 'DiscourseConnect provider' authentication. If `enforce_second_factor_on_external_auth` is enabled, users will not be able to log in with external authentication providers after they set up two-factor authentication.